Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth+OtherAuthActions.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+import MongoSwift
+
+extension CoreStitchAuth {
+    /**
+     * Logs out the current user, and clears authentication state from this `CoreStitchAuth` as well as underlying
+     * storage. Blocks the current thread until the request is completed. If the logout request fails, this method will
+     * still clear local authentication state.
+     */
+    public func logoutInternal(withUserId userId: String?) throws {
+        objc_sync_enter(authOperationLock)
+        defer { objc_sync_exit(authOperationLock) }
+
+        // Unwrap the userId or use the active user
+        guard let userId = userId != nil ? userId : activeUserAuthInfo?.userID else {
+            return
+        }
+
+        // Get the user objects if it exists --> otherwise throw
+        let (index, authInfo) = try getUserAndIndexOrThrow(withId: userId)
+
+        // If the user is not logged in --> return
+        guard authInfo.isLoggedIn else { return }
+
+        // If it is an AnonymousUser remove the user altogether from the list
+        // Otherwise set the authInfo to have no access token or refresh token
+        let newAuthInfo = StoreAuthInfo.init(withAuthInfo: authInfo, withLogout: true).toAuthInfo
+        if authInfo.loggedInProviderType == .anonymous {
+            loggedInUsersAuthInfo.remove(at: index)
+        } else {
+            loggedInUsersAuthInfo[index] = newAuthInfo
+        }
+
+        // TODO: what should happen if we fail here?
+        try writeCurrentUsersAuthInfoToStorage(
+            currentUsersAuthInfo: loggedInUsersAuthInfo,
+            toStorage: storage)
+
+        // If this was the active user then set the activeUser to nil and clear its authState
+        if userId == activeUserAuthInfo?.userID {
+            // This will call onAuthEvent
+            clearActiveUser(storage: storage)
+        }
+
+        // Issue the logout request
+        _ = try? self.doLogout(authInfo)
+    }
+
+    /**
+     * Leaves the active user as logged in but makes the user with the given id the active user
+     */
+    public func switchToUserWithIdInternal(_ userId: String) throws -> TStitchUser {
```

Comment:
[style] This method should be renamed to resspond with its parameter, e.g., `switchToUserInternal(withId:)`